benchmarks/nHcal/basic_distribution/Snakefile

```diff
@@ -0,0 +1,55 @@
+import os
+
+DETECTOR_PATH = os.environ["DETECTOR_PATH"]
+
+rule nhcal_basic_distribution_simulate:
+    output:
+        "sim_output/nhcal_basic_distribution/E{ENERGY}GeV/sim_{DETECTOR_CONFIG}.{INDEX}.edm4hep.root",
+    params:
+        N_EVENTS=10000,
+    shell:
+        """
+set -m
+exec npsim \
+    --compactFile {DETECTOR_PATH}/{wildcards.DETECTOR_CONFIG}.xml \
+    --numberOfEvents {params.N_EVENTS} \
+    --random.seed $RANDOM \
+    --enableGun \
+    -v WARNING \
+    --gun.particle neutron \
+    --gun.thetaMin 120*degree \
+    --gun.thetaMax 180*degree \
+    --gun.distribution uniform \
+    --gun.energy "{wildcards.ENERGY}*GeV" \
+    --outputFile {output}
+"""
+
+
+rule nhcal_basic_distribution_combine:
+    input:
+        lambda wildcards: expand(
+            os.path.join("sim_output/nhcal_basic_distribution/E{ENERGY:.1f}GeV/", "sim_{DETECTOR_CONFIG}.{INDEX:02d}.edm4hep.root"), 
+            DETECTOR_CONFIG=wildcards.DETECTOR_CONFIG,
+            ENERGY=float(wildcards.ENERGY),
+            INDEX=range(int(wildcards.N)),
+        ),
+    wildcard_constraints:
+        N=r"\d+",
+        ENERGY=r"\d+(\.\d+)?"
+    output:
+        f"sim_output/nhcal_basic_distribution/sim_{{DETECTOR_CONFIG}}_E{{ENERGY}}GeV_combined_{{N}}files.edm4hep.root",
+    shell:
+        """
+hadd {output} {input} 
```

Comment:
Unfortunately, the output files are a bit too large. Any ideas on how we could reduce them?
https://eicweb.phy.anl.gov/EIC/benchmarks/detector_benchmarks/-/jobs/5460747#L1074

Not doing hadd and pass a list to TChain would give a factor of 2. Maybe you don't need to do npsim, can do just do ddsim (main difference is no optical photons in pfRICH).

Reply:
What do you mean and how are you able to check it? We do need the output from npsim or ddsim so these files will be created anyway. We can skip hadd and just use the list. I assume that edm4hep files are removed automatically or can we just delete them after analysis step?

Reply:
I spoke to Dima and the answer is we just have to make sure we cleanup the files. Please remove the files once they are no longer needed. This means we need to analyze the files and remove them afterwards.